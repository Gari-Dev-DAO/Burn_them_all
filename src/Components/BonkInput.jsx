
import { BONK_LIMIT } from "../utils/BonkAdmin"
import Input from "./Input"

const BonkInput = ({bonkAmount,setBonkAmount}) => {
 
  return (
    <div className="bonkInput">
    <p className="bonkAmount">AMOUNT OF BONKS</p>
    <Input value={bonkAmount} setValue={setBonkAmount}/>
    <p className="bonkLimit">{BONK_LIMIT} BONK $</p>
    </div>
  )
}

export default BonkInput