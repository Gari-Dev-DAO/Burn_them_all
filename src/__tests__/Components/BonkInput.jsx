import { render } from "@testing-library/react";
import BonkInput from "../../Components/BonkInput";

let mockbonkAmount = { bonkAmount: 1 };
let mockupdateBonkAmount = { updateBonkAmount: jest.fn() };

jest.mock("react", () => {
  const ActualReact = jest.requireActual("react");
  return {
    ...ActualReact,
    useContext: () => [mockbonkAmount],
  };
});

jest.mock("../../hooks/useDispatcher", () => ({
  useDispatcher: () => {
    return mockupdateBonkAmount;
  },
}));

describe("Bonk Input", () => {
  it("renders component", () => {
    const { container } = render(<BonkInput />);
    expect(container).toMatchSnapshot();
  });
});
