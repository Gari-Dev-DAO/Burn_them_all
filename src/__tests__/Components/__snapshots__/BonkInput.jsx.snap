// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bonk Input renders component 1`] = `
<div>
  <div
    class="bonkInput"
  >
    <p
      class="bonkAmount"
    >
      AMOUNT OF BONKS
    </p>
    <div
      style="display: flex; justify-content: center; align-items: center; flex-direction: column; gap: 5px;"
    >
      <button
        data-testid="button"
        style="background-color: transparent; cursor: pointer; z-index: 200;"
      >
        <svg
          color="#FED900"
          fill="currentColor"
          height="25px"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(254, 217, 0);"
          viewBox="0 0 1024 1024"
          width="25px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
          />
        </svg>
      </button>
      <input
        aria-label="bonk-input"
        class="input"
        type="number"
        value="1"
      />
      <button
        data-testid="button"
        style="background-color: transparent; cursor: pointer; z-index: 200;"
      >
        <svg
          color="#FED900"
          fill="currentColor"
          height="25px"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(254, 217, 0);"
          viewBox="0 0 1024 1024"
          width="25px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </button>
    </div>
    <p
      class="bonkLimit"
    >
      100
       BONK $
    </p>
  </div>
</div>
`;
